components/tcf/secondLayer/src/components/PurposeGroup.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import SuiSwitch from '@s-ui/react-atom-switch'
+
+export default function PurposeGroup({
+  name,
+  descriptions,
+  descriptionField,
+  state,
+  onConsentsChange,
+  onLegitimateInterestsChange
+}) {
+  const handleConsentToggle = ({index, value}) => {
+    onConsentsChange({index, value})
+  }
+
+  const handleLegitimateInterestsToggle = ({index, value}) => {
+    onLegitimateInterestsChange({index, value})
+  }
```

Comment:
This 2 functions are redundant. You can use the onConsentChange/onLegitimateInteresesChange in the onToggle events